zstacklib/zstacklib/utils/gpu.py

```
@@ -561,6 +561,49 @@ def get_huawei_gpu_aios_rank_table_dict(npu_ids, iswindows=False):
 
 
+def check_huawei_npu_is_isolated(npu_id, all_npu_ids, iswindows=False):
+    """
+    Check whether a Huawei NPU is isolated using `npu-smi info -t hccs`.
+    Return True when health status is not OK. Do not inspect lane majority.
+    """
+    if not npu_id or not all_npu_ids or len(all_npu_ids) <= 1:
+        return False
+
+    try:
+        r, _, _ = bash_roe("which npu-smi")
```

Ruff (0.14.7)

`bash_roe` may be undefined, or defined from star imports

(F405)